### PR TITLE
Dispatch tests and metrics flag

### PR DESCRIPTION
**Pull request by josephschorr**

### CODE REVIEW

File: internal/datastore/mysql/options.go [hunk 1]
```diff
@@ -39,14 +39,15 @@ type Option func(*mysqlOptions)
...
-		splitAtUsersetCount: defaultUsersetBatchSize,
+		gcWindow:              defaultGarbageCollectionWindow,
+		gcInterval:            defaultGarbageCollectionInterval,
+		gcMaxOperationTime:    defaultGarbageCollectionMaxOperationTime,
+		watchBufferLength:     defaultWatchBufferLength,
+		maxOpenConns:          defaultMaxOpenConns,
+		connMaxIdleTime:       defaultConnMaxIdleTime,
+		connMaxLifetime:       defaultConnMaxLifetime,
+		splitAtUsersetCount:   defaultUsersetBatchSize,
+		enablePrometheusStats: false,
```
Comment (vroldanbet) [inline internal/datastore/mysql/options.go:50]: nit: do we want to pull a constant just like every other option?

File: internal/datastore/mysql/options.go [hunk 2]
```diff
@@ -103,12 +104,13 @@ func TablePrefix(prefix string) Option {
...
 }
 
-// EnablePrometheusStats enables Prometheus metrics provided by Go's database/sql package
+// WithEnablePrometheusStats marks whether Prometheus metrics provided by Go's database/sql package
+// are enabled.
 //
-// Prometheus metrics are enabled by default.
-func EnablePrometheusStats() Option {
+// Prometheus metrics are disabled by default.
+func WithEnablePrometheusStats(enablePrometheusStats bool) Option {
```
Comment (vroldanbet) [inline internal/datastore/mysql/options.go:111]: naming nit: perhaps `WithPrometheusStats` would be easier to reason about?

File: pkg/cmd/datastore/datastore.go [hunk 3]
```diff
@@ -104,6 +104,7 @@ func RegisterDatastoreFlags(cmd *cobra.Command, opts *Config) {
 	cmd.Flags().DurationVar(&opts.RequestHedgingInitialSlowValue, "datastore-request-hedging-initial-slow-value", 10*time.Millisecond, "initial value to use for slow datastore requests, before statistics have been collected")
 	cmd.Flags().Uint64Var(&opts.RequestHedgingMaxRequests, "datastore-request-hedging-max-requests", 1_000_000, "maximum number of historical requests to consider")
 	cmd.Flags().Float64Var(&opts.RequestHedgingQuantile, "datastore-request-hedging-quantile", 0.95, "quantile of historical datastore request time over which a request will be considered slow")
+	cmd.Flags().BoolVar(&opts.EnableDatastoreMetrics, "datastore-prometheus-metrics", true, "set to false to disabled prometheus metrics from the datastore")
```
Comment (vroldanbet) [inline pkg/cmd/datastore/datastore.go:107]: this does not seem to align with the default in the individual stores? e.g. https://github.com/authzed/spicedb/pull/561/files#diff-1f9d83984c96749a94659b71f4b5b8366b0b98e11e71923100983d110cd7b3a3R50

File: internal/datastore/mysql/options.go (hunk 1 above)
Comment (josephschorr) [reply internal/datastore/mysql/options.go:50]: Changed

File: internal/datastore/mysql/options.go (hunk 2 above)
Comment (josephschorr) [reply internal/datastore/mysql/options.go:111]: I vacillated back and forth on that, but I decided to be explicit

File: pkg/cmd/datastore/datastore.go (hunk 3 above)
Comment (josephschorr) [reply pkg/cmd/datastore/datastore.go:107]: Correct. It should be enabled by default from the CLI, but disabled by default if you just instantiate the datastore directly

Comment (vroldanbet) [reply pkg/cmd/datastore/datastore.go:107]: got it, it's intentional 👍🏻

**Review by vroldanbet — APPROVED** [review]

**Review by ecordell — DISMISSED** [review]
LGTM

**Review by ecordell — APPROVED** [review]

**Review by ecordell — APPROVED** [review]

**Pull request merged**